### PR TITLE
inverse_qft added

### PR DESCRIPTION
**Pull request by mmvandieren**
### Description

This is related to [Issue 1783](https://github.com/NVIDIA/cuda-quantum/issues/1783) as it provides an example missing from the documentation about how to apply the adjoint operator.  It also demonstrates the variety of ways that one can chose to define a kernel, and is important for those that want to be able to apply QFT to a register within a larger quantum algorithm, which is often the use case.  Finally, this example will be referred to in a future notebook on Shor's algorithm.

### CODE REVIEW

File: docs/sphinx/examples/python/tutorials/quantum_fourier_transform.ipynb [hunk 1]
```diff
@@ -251,6 +251,89 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
```
Comment (bmhowe23) [inline docs/sphinx/examples/python/tutorials/quantum_fourier_transform.ipynb:273]: The other cells' outputs in this notebook are already filled out (as if the notebook was already run). If you want the new cells to look the same in the published documentation, I think you need to run them locally and then commit the results into your PR. Is that what you, or do you want the published docs to leave these cells' output as empty?

Comment (mmvandieren) [reply docs/sphinx/examples/python/tutorials/quantum_fourier_transform.ipynb:273]: I'll run them and recommit.  thanks

Comment (mmvandieren) [reply docs/sphinx/examples/python/tutorials/quantum_fourier_transform.ipynb:273]: done

File: docs/sphinx/examples/python/tutorials/quantum_fourier_transform.ipynb [hunk 2]
```diff
@@ -251,6 +251,115 @@
...
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Quantum Fourier transform revisited\n",
+    "We provided one implementation of the quantum Fourier Transform above.  To demonstrate some of the other features of CUDA-Q, let's define a new kernel for the Quantum Fourier Transform, which we'll call `quantum_fourier_transform2`.  Pay attention to how the kernel definition differs from the example above.\n",
+    "\n",
+    " Since the quantum Fourier Transform is a unitary operator, its inverse is its adjoint.  Therefore, once we have implemented the quantum Fourier transform as a kernel in CUDA-Q which acts, we can use the built-in `adjoint` operation to create the inverse quantum Fourier transform."
```
[the remark this answers is not in the document]
Comment (mmvandieren) [reply docs/sphinx/examples/python/tutorials/quantum_fourier_transform.ipynb:261]: ah, yes.  "which acts" should be removed. I was having difficulty trying to explain how this kernel differed from the one earlier defined in terms of what it takes as an argument. I'll commit that.

[the remark this answers is not in the document]
Comment (mmvandieren) [reply docs/sphinx/examples/python/tutorials/quantum_fourier_transform.ipynb]: fixed the "which acts" phrase

**Review by bmhowe23 — APPROVED** [review]
LGTM. Thanks, Monica.

**Pull request merged**